libbpf-tools/trace_helpers.h

```diff
@@ -43,4 +43,34 @@ int bump_memlock_rlimit(void);
 
 bool is_kernel_module(const char *name);
 
+/*
+ * When we need for kprobe/kretprobe, please check out new fentry/fexit
+ * probes, as they provide better performance and usability. But in some
+ * situations we have to fallback to kprobe/kretprobe probes. This helper
+ * is used to detect those ones before attaching.
+ *
+ *	1. A gap between kernel versions, kernel BTF is exposed
+ * 	   starting from 5.4 kernel. but fentry/fexit is actually
+ * 	   supported starting from 5.5.
+ *	2. Wether kernel support module BTF or not
+ *
+ * *name* is the name of a function to be attached, which can be from
+ * vmlinux or a kernel module.
+ * *mod* is the name of a kernel module, if NULL, it means *name*
+ * belongs to vmlinux.
+ */
+bool fentry_exists(const char *name, const char *mod);
+
+/*
+ * The name of a function to be attached may be changed between kernel
```

Comment:
same, this is not a name of a function we are attaching, it's the name of a function we are attaching *to*. So, as above, be more specific: "the name of a kernel function to be attached to"